From 4c95454cd8f8ebdaf065b99ec89c7b0da743df37 Mon Sep 17 00:00:00 2001
From: zwx1052249 <zhouwenpei1@Huawei.com>
Date: Tue, 27 Jul 2021 10:15:46 +0800
Subject: [PATCH] fix build fail caused by file format not recognized

---
 .../libeppic/Makefile                                           | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eppic-d84c3541035d95077aa8571f5d5c3e07c6ef510b/libeppic/Makefile b/eppic-d84c3541035d95077aa8571f5d5c3e07c6ef510b/libeppic/Makefile
index 8b97c87..1debebe 100644
--- a/eppic-d84c3541035d95077aa8571f5d5c3e07c6ef510b/libeppic/Makefile
+++ b/eppic-d84c3541035d95077aa8571f5d5c3e07c6ef510b/libeppic/Makefile
@@ -104,7 +104,7 @@ default: $(TARGETS)
 $(CFILES): $(HFILES) eppic.tab.h
 
 $(TARGETS): $(OFILES)
-	$(AR) ccurl $(TARGETS) $(OFILES)
+	$(AR) rs $(TARGETS) $(OFILES)
 
 clean: 
 	-/bin/rm -f *.o $(TARGETS) $(LDIRT)
-- 
2.27.0

